app/controllers/facebook_controller.rb

```diff
@@ -0,0 +1,9 @@
+class FacebookController < ApplicationController
+  def create
+    user = User.create_with_omniauth(request.env['omniauth.auth'])
+    user.save!
+    session[:user_id] = user.id
```

Comment:
I think this was the root of the problem. We were trying to add the user's id to the session before actually saving the user to the db. I am hoping that this will fix our problem in production.

Reply:
Ooo good catch

Reply:
Nice!!!! I hope this does the trick.  Good thinking.

Reply:
Nice Work!